src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurveyDao.java

```diff
@@ -0,0 +1,365 @@
+import org.sagebionetworks.bridge.models.GuidCreatedOnVersionHolder;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifierImpl;
+import org.sagebionetworks.bridge.models.surveys.Survey;
+import org.sagebionetworks.bridge.models.surveys.SurveyElement;
+import org.sagebionetworks.bridge.models.surveys.SurveyElementFactory;
+import org.sagebionetworks.bridge.models.surveys.SurveyId;
+import org.sagebionetworks.bridge.models.upload.UploadSchema;
+import org.sagebionetworks.bridge.services.UploadSchemaService;
+import org.sagebionetworks.bridge.time.DateUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+public class HibernateSurveyDao implements SurveyDao {
+    private static final String IDENTIFIER_PREFIX = "identifier:";
+    Integer OFFSET = 0;
+    Integer PAGESIZE = 1;
+    
+    private static final String SELECT_TEMPLATE = "SELECT survey ";
+    private static final String SELECT_ELEMENT_TEMPLATE = "SELECT surveyElement ";
+    
+    private static final String GET_ALL = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND guid = :guid";
+    
+    private static final String GET_ALL_SURVEYS = "FROM HibernateSurvey as survey WHERE studyKey = :studyKey";
+    
+    private static final String GET_ALL_ACTIVE_PUBLISH = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND deleted = 0 AND published = 1 ORDER BY createdOn DESC";
+    
+    private static final String GET_ACTIVE = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND guid = :guid AND deleted = 0 ORDER BY createdOn DESC";
+    
+    private static final String GET_ACTIVE_PUBLISH = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND guid = :guid AND deleted = 0 AND published = :published ORDER BY createdOn DESC";
+    
+    private static final String GET_ACTIVE_PUBLISH_WITH_IDENTIFIER = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND identifier = :identifier AND deleted = 0 AND published = :published ORDER BY createdOn DESC";
+    
+    private static final String GET_ACTIVE_WITH_IDENTIFIER = "FROM HibernateSurvey as survey " + 
+            "WHERE studyKey = :studyKey AND identifier = :identifier AND createdOn = :createdOn AND deleted = 0 ORDER BY createdOn DESC";
+    
+    private static final String GET_ELEMENTS = "FROM HibernateSurveyElement as surveyElement " + 
+            "WHERE surveyGuid = :surveyGuid AND createdOn = :createdOn ORDER BY createdOn DESC";
+    
+    private static final String DELETE_SURVEY_ELEMENTS = "DELETE FROM HibernateSurveyElement WHERE surveyGuid = :surveyGuid AND createdOn = :createdOn";
```

Comment:
For the queries that get one survey guid or identifier, we want to sort on createdOn desc.

For the queries that get all surveys in a study, we want to sort on survey guid (or identifier), and then sort on createdOn desc.